### DISCUSSION TITLE
Pulsate のリリースサイクルについて

### DISCUSSION THREAD

**Discussion by m1sk9** [pulsate-dev/pulsate#650] [General]
現状 Pulsate のリリースサイクルというのは決まっておらず, @m1sk9 の声でリリースというのでおそらく今後続くであろうと思われるが, Pulsate （Misskey も Mastodon も） は変化の激しいプロジェクトなのでリリースというのはやはり CI/CD の面で自動化したくリリースサイクルというのは明確にしておきたい.

**Comment by m1sk9** [comment]
私がパッと思いつくのは Firefox や Thunderbird が採用してる **4週間ごとの[ラピッドリリース](https://ja.m.wikipedia.org/wiki/Mozilla_Firefox#%E3%83%AA%E3%83%AA%E3%83%BC%E3%82%B9%E3%82%B5%E3%82%A4%E3%82%AF%E3%83%AB)** 

> Firefox 5から6の間は例外的に8週間だが、Firefox 6以降は基本6週間ごとに最新版がリリースされ、2011年中にはFirefox 9までアップデートされた。2013年はFirefox 26まで、2014年はFirefox 34まで、2015年はFirefox 43までリリースされた。 2016年より6〜8週間の間隔での休日に合わせた不規則なリリーススケジュールとなり、2019年後半からは4週間毎のリリースとなっている。

これに合わせて Nightly (アルファ版) を導入して一部のユーザーに使ってもらいバグ報告などを行ってもらいなるべく安定版にバグを持ち込むことを少なくしたい

**Reply by laminne** [reply]
個人的にはいつまでにリリースというのはあまりすべきではないという考えです  
こんな感じにしたら良さそう:
- Nightly(alpha): mainブランチ, 最も最新のブランチ. 実行 or ビルドが確実に通る保証はない
- Beta(staging): stagingブランチ, 実行とビルドが確実に動き、pulsate-devのステージング環境に毎回デプロイされる
  - Betaで最短3日テストしてからStableにPRを立てる
- Stable: releaseブランチ, 正式リリース版

**Reply by Allianaab2m** [reply]
Stable = mainブランチの方がいいような気がします

**Reply by m1sk9** [reply]
まあそうだよね. どっちかというと Nightly に相当するのが `develop` だと思う

**Reply by laminne** [reply]
それはそうとしてどれをdefaultブランチに指定するか問題が発生しそう

**Reply by m1sk9** [reply]
`main` では...???

**Comment by laminne** [comment]
リリース自動化やそれに伴う設定などについても話し合う必要がありそう
